packages/jellyfish-api-core/__tests__/category/icxorderbook/createOrder.test.ts

```diff
@@ -0,0 +1,174 @@
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import {
+  ICXGenericResult, ICXOfferInfo, ICXOrderInfo, ICXOrder, InputUTXO
+} from '../../../src/category/icxorderbook'
+import BigNumber from 'bignumber.js'
+import { accountDFI, checkBTCSellOrderDetails, checkDFISellOrderDetails, idDFI, setup } from './common.test'
+import { RpcApiError } from '../../../src'
+
+describe('Should test ICXOrderBook.createOrder', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+    await setup(container)
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  afterEach(async () => {
+    // cleanup code here
+  })
+
+  it('Should create an ICX order to sell 15 DFI', async () => {
+    const accountDFIStart = await container.call('getaccount', [accountDFI, {}, true])
+    // create order - maker
+    const order: ICXOrder = {
+      tokenFrom: idDFI,
+      chainTo: 'BTC',
+      ownerAddress: accountDFI,
+      receivePubkey: '037f9563f30c609b19fd435a19b8bde7d6db703012ba1aba72e9f42a87366d1941',
+      amountFrom: new BigNumber(15),
+      orderPrice: new BigNumber(0.01)
+    }
+    const result: ICXGenericResult = await client.icxorderbook.createOrder(order, [])
+    const createOrderTxId = result.txid
+    await container.generate(1)
+
+    // list ICX orders
+    const orders: Record<string, ICXOrderInfo| ICXOfferInfo> = await container.call('icx_listorders', [])
+    // we know that only ICXOrderInfo will be returned, so cast and pass to check order details
+    await checkDFISellOrderDetails(container, order, createOrderTxId, orders as Record<string, ICXOrderInfo>)
```

Comment:
can we have all the assertions directly here? it's easier to read what's going on.

Reply:
okay, this is a kind of tradeoff. I would also prefer to have checks inside the test segment itself. But sometimes it adds more clutter. especially if the checks belong to the setup code. making it harder to read what that specific test is about.
we have  6 functions like this that just check whether the passed(previously created) transaction is there in the system.
```
checkDFISellOrderDetails()
checkBTCSellOrderDetails()
checkDFIBuyOfferDetails()
checkBTCBuyOfferDetails()
checkDFCHTLCDetails()
checkEXTHTLCDetails()
```
What each function does is described in the function name

Reply:
in this case, you should maintain a list of expects

> https://jestjs.io/docs/expect

Reply:
are you saying to use a matcher like `.toMatchObject(object)` instead of the functions above? or something else?